pages/index.tsx

```diff
@@ -412,20 +456,27 @@ export default function Home({ featuredStorefronts }: HomeProps) {
                       scrollToListingTop();
                     }}
                   >
-                    {sortOptions[filterBy]
-                      .map(({ label, key }) => (
-                        <Option key={key} value={key}>
-                          {label}
-                        </Option>
-                      ))}
+                    {sortOptions[filterBy].map(({ label, key }) => (
+                      <Option key={key} value={key}>
+                        {label}
+                      </Option>
+                    ))}
                   </SelectInline>
                 </Space>,
               ]}
             />
             <Row gutter={24}>
-              {take(show, displayedListings).map((listing: Listing) => (
+              {take(show, displayedListings).map((listing: Listing, i) => (
                 <Col xs={24} sm={12} md={8} lg={8} xl={6} xxl={6} key={listing?.listingAddress}>
-                  <ListingPreview {...listing} />
+                  <ListingPreview
+                    listing={listing}
+                    meta={{
+                      index: i,
+                      list: 'current-listings',
+                      sortBy: sortBy,
+                      filterBy: filterBy,
```

Comment:
Instead of passing sortBy and filterBy to each preview look to decorate a track function in this view and pass that to preview.

```
const listingTrack = (args...) => track({...args, sortBy, filterBy })
```

Reply:
Yeah, that's a good idea. Will do!

Do you mind if we keep passing down listing as a named prop though? I think it's cleaner than the {... listing} syntax :shrug: